### PR TITLE
Support noop-ing IntervalScan runs

### PR DESCRIPTION
**Pull request by jcorbin**
r @raynos @rf

### CODE REVIEW

File: lib/interval-scan.js [hunk 1]
```diff
@@ -135,7 +135,11 @@ function run(callback) {
 
     self.clearTimer();
 
-    self.theRun = new IntervalScanRun(self, runDone);
+    var collection = self.getCollection();
+    var keys = collection && Object.keys(collection);
```
Comment (Raynos) [inline lib/interval-scan.js]: consider fast check for array...

**Comment by Raynos** [comment]
lgtm.

**Comment by rf** [comment]
:+1:

File: lib/interval-scan.js [hunk 2]
```diff
@@ -135,7 +135,16 @@ function run(callback) {
 
     self.clearTimer();
 
-    self.theRun = new IntervalScanRun(self, runDone);
+    var collection = self.getCollection();
+    var keys = null;
+    if (Array.isArray(collection)) {
+        keys = collection.length;
```
Comment (Raynos) [inline lib/interval-scan.js]: `keysLength`

**Comment by Raynos** [comment]
legit. +1

**Pull request merged**